python/hail/methods/statgen.py

```diff
@@ -1635,7 +1650,9 @@ def split_multi_hts(ds, keep_star=False, left_aligned=False) -> MatrixTable:
     if 'PID' in entry_fields:
         update_entries_expression['PID'] = split.PID
 
-    return split.annotate_entries(**update_entries_expression).drop('old_locus', 'old_alleles')
+    return split.annotate_rows(
+        **update_rows_expression).annotate_entries(
```

Comment:
also this will invalidate the annotate entries

Reply:
need to use `_annotate_all` I think

Reply:
nifty!